.github/bin/download-maven-dependencies.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Enable all known profiles to make sure their plugin dependencies are downloaded
+# GIB should be disabled even though it's profile is activated, to make sure it doesn't skip any submodules
+$RETRY $MAVEN -P ci,errorprone-compiler ${MAVEN_GIB} -Dgib.disable dependency:go-offline
+$RETRY $MAVEN -P ci,errorprone-compiler ${MAVEN_GIB} -Dgib.disable de.qaware.maven:go-offline-maven-plugin:resolve-dependencies
```

Comment:
reading `de.qaware.maven:go-offline-maven-plugin:resolve-dependencies`'s docs i am under impression it wants to be a better drop-in replacement for `dependency:go-offline`.
do we need both?

Reply:
> reading `de.qaware.maven:go-offline-maven-plugin:resolve-dependencies`'s docs i am under impression it wants to be a better drop-in replacement for `dependency:go-offline`. do we need both?

We need both - one fixes the problems with the other.

Reply:
i am looking at https://github.com/qaware/go-offline-maven-plugin README and while it discusses `dependency:go-offline` existence, it purports it fixes problems with it, and doesn't advise to call both.

What is the issue with `de.qaware.maven:go-offline-maven-plugin:resolve-dependencies` that renders it insufficient? did we choose to report it? It would be nice to have a TODO comment here indicating when one can be removed

Reply:
The issue was described by the author here: https://github.com/qaware/go-offline-maven-plugin/issues/23#issuecomment-739778900, but the original issue only mentioned plugins, while I believe this affects all collected dependencies. The fix introduced in `1.2.8` only changed `Set<Plugin>` into `List<Plugin>`, without changing all the other places where usage of `Set` might remove duplicates. 

I've left a comment in the original issue thread.

I will report a separate issue and add a TODO here.